### PR TITLE
Enabled wildcard default includes (*) to retrieve inherited object properties (Removing ACF 11 Compatibility)

### PR DESCRIPTION
**Pull request by homestar9**
Previously if you wanted your mementified objects to retrieve inherited properties from a base class, you couldn't use `defaultIncludes = [ "*" ]`.   In this update, the mementifier will scan all inherited properties and apply them before evaluating any `defaultExcludes`.

I also tweaked the model a little bit to showcase the ability to retrieve inherited properties using a wildcard `defaultIncludes`.   Even with this minor change, the original tests continue to pass.  I added a new test to make sure the returned memento includes base properties. This should enable users to write less code and still have the flexibility to customize your mementos.  It also helps your base classes from having to keep track of which properties it should provide to its children.

Finally, I also removed the .env file from the repository (contributors should use the .env.template file instead).  I updated the .env.template and .cfconfig file to allow a few additional properties in case you want to test with different database engines (e.g. MSSQL server)

### CODE REVIEW

**Comment by lmajano** [comment]
@homestar9 Great job! The only issue, is that all the tests failed!

**Comment by lmajano** [comment]
Can you please revise them so I can merge?

**Comment by homestar9** [comment]
That is so strange! They all passed on my end.  I wonder if something in the Travis CI config needs to be updated because I made changes to the .env template and removed .env from the repo?  This is the error I see when I look at the details:

`No suitable driver available for mementifier, please check the driver setting in resources file`

**Comment by homestar9** [comment]
Update: It looks like removing the .env file was the culprit.   When I put it back, the Adobe 2016 and 2018 tests passed. However, there still seems to be a problem with Lucee 5:
`Could not create connection to database server.`

The Adobe 2011 issue appears to be unrelated to any changes I made as far as I can tell.  It looks like it doesn't know how to execute `keyExists()` in a few places.

I am reverting the .env, travis, and server config to their original state locking things into MySQL to see if that fixes things.

**Comment by homestar9** [comment]
Updated: Locking things into MySQL did fix the Travis build tests.

I went into `mementifier.cfc` and made some adjustments for ACF 2011 compatibility. Hopefully this should do the trick.

**Comment by homestar9** [comment]
Fixed ACF 2011 compatibility. Keeping my fingers crossed that the tests pass this time.

**Comment by homestar9** [comment]
I did a little digging as to what was causing problems with ACF 11. Here's what I discovered:

When you attempt to merge a component's `metadata.properties` "array" into another array like this:
`properties.append( arguments.metadata.properties, true );`
It doesn't respect the "merge" argument. Here's the result:
https://i.imgur.com/3v2Fri7.png

However, running `arrayLen( arguments.metadata.properties )` reveals a value of 3 and `isArray( arguments.metadata.properties )` returns a value of yes.

If you attempt to serialize and deserialize the array, things work as you would expect:
`properties.append( deSerializeJson ( serializeJson( arguments.metadata.properties ) ) )`
https://i.imgur.com/Rt40M5O.png

Based on my findings, ACF 11 treats `metadata.properties` of a component as something other than a traditional CF array, which is why it cannot be appended with the merge argument set to true.

This bug doesn't exist in ACF 2016, 2018, or Lucee 5.

In the meantime, the only way to work around the issue is to not use the member function `append()` and instead do a simple `for` loop to append the values we need.  Once ACF 11 support can be dropped from Mementifier, we can switch to using the member functions.

**TLDR:** 
ACF 11 treats a component's `metadata.properties` as a non-traditional array so it can't be appended to another array with `merge` set to true.

**Comment by lmajano** [comment]
@homestar9 WOW you have done incredible!  I say forget ACF11, we are not supporting it in many of our modules now.  So forget about ACF11. Once the rest of the tests pass, I will bump to a major version

**Comment by homestar9** [comment]
Thank you @lmajano, I'm glad you are happy with the changes.  I reverted the ACF 11 tweaks so things should be all good to go now.  Have a great Thanksgiving!

**Comment by homestar9** [comment]
It looks like all tests passed except for ACF 11.
image

**Comment by lmajano** [comment]
Thanks! I will work on this, this week.

**Comment by homestar9** [comment]
Happy new year!
Just a reminder that the failing test for ACF 11 is intentional.  All other tests have passed.

**Comment by lmajano** [comment]
Thanks, coming back from PTO!!!

**Comment by homestar9** [comment]
Nice! I hope you had a relaxing time off!
